master/internal/api_workspace.go

```diff
@@ -521,6 +588,128 @@ func (a *apiServer) PatchWorkspace(
 		errors.Wrapf(err, "error refetching updated workspace (%d) from db", currWorkspace.Id)
 }
 
+func (a *apiServer) modifyWorkspaceNamespaceBinding(ctx context.Context,
+	req *apiv1.ModifyWorkspaceNamespaceBindingRequest, tx *bun.Tx, w *model.Workspace) (*apiv1.ModifyWorkspaceNamespaceBindingResponse, error) {
+	if req.AutoCreateNamespace {
+		license.RequireLicense("auto-create namespace")
+		if req.NamespaceName != nil {
+			return nil, status.Error(codes.InvalidArgument, "Cannot specify a namespace if you want to auto-create")
+		}
+		namespaceName, err := generateNamespaceName(w.Name)
+		if err != nil {
+			return nil, errors.Wrapf(err, "error creating namespace %s", *namespaceName)
+		}
+		req.NamespaceName = namespaceName
+	}
+
+	// Create the namespace in Kubernetes.
+	err := a.m.rm.CreateNamespace(req.AutoCreateNamespace, *req.NamespaceName, req.ClusterName)
+	if err != nil {
+		return nil, fmt.Errorf("error creating k8s namespace: %w", err)
+	}
+
+	var wsns model.WorkspaceNamespace
+
+	// If there is more than one namespace bound to a workspace within a given cluster, this query
+	// should fail (because the result would have type []int and []string rather than int and string
+	// string, respectively).
+	err = tx.NewSelect().Model(&model.WorkspaceNamespace{}).
+		Where("workspace_id = ?", w.ID).
+		Where("cluster_name = ?", req.ClusterName).
+		Scan(ctx, &wsns)
```

Comment:
same as above, put all tx in `Bun().RunInTx(...)`

Reply:
This takes the transaction passed into the function, it doesn't start a new one. 
However, I changed the functions that call `modifyWorkspaceNamespaceBinding` to use `RunInTx`rather than `BeginTx`